app/controllers/users_controller.rb

```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+
+class UsersController < ApplicationController
+  # if a user is authenticated then it will redirect to the root path
+  before_action :redirect_if_authenticated, only: %i[create new]
+  # if a user is authenticated then he/she can access edit destroy or update
+  # action otherwise it will redirect to the login path
+  before_action :authenticate_user!, only: %i[edit destroy update edit_password change_password]
+  def new
+    @user = User.new
+  end
+
+  def destroy
+    current_user.destroy
+    reset_session
+    redirect_to root_path, notice: 'This user account has been deleted'
+  end
+
+  def edit_password
+    # assign current object property
+    # because we did not available all of user model data into view
+    # this reason i made a user_view_model and assign specific property and send to view
+    @user = UserViewModel.new(email: current_user.email, current_password: '', password: '', password_confirmation: '')
+  end
+
+  def change_password
+    @user = current_user
+    if @user.authenticate(params[:user][:current_password])
+      if @user.update(password: params[:user][:password], password_confirmation: params[:user][:password_confirmation])
+        redirect_to root_path, notice: 'Password updated'
+      else
+        redirect_to change_password, status: :unprocessable_entity
+      end
+    else
+      flash[:error] = 'Incorrect password'
+      redirect_to change_password, status: :unprocessable_entity
+    end
+  end
+
+  def update
+    @user = current_user
+    # assign current object property
+    # because we did not available all of user model data into view
+    # this reason i made a user_view_model and assign specific property and send to view
```

Comment:
As this is a demo project, if you are storing the comments as a note, than all okay keep it, otherwise we do not use this kind of comment in our projects. Thank you!